src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,855 @@
+
+            ContinuousRead = true;
+
+            long samplePerTicks = Stopwatch.Frequency / frequency;
+            if (samplePerTicks == 0)
+            {
+                samplePerTicks = 1;
+            }
+
+            _stopWatch.Restart();
+            Task samplingTask = Task.Run(() =>
+            {
+                long samples = 0;
+                long nextSampleAt = _stopWatch.ElapsedTicks;
+                long elapsedTicks = 0;
+                long jitter = 0;
+                long maxJitterCorrectionPerSample = Math.Max(Stopwatch.Frequency / 100000, 1);
+                while (_stopWatch.IsRunning)
+                {
+                    elapsedTicks = _stopWatch.ElapsedTicks;
+                    if (elapsedTicks >= nextSampleAt)
+                    {
+                        nextSampleAt = elapsedTicks + samplePerTicks;
+                        ReadADCValue();
+                        samples++;
+                        jitter = (elapsedTicks - (samples * samplePerTicks));
+                        if (JitterCorrection)
+                        {
+                            if (jitter > 0)
+                            {
+                                nextSampleAt -= Math.Min(jitter, maxJitterCorrectionPerSample);
+                            }
+                            else if (jitter < 0)
+                            {
+                                nextSampleAt += Math.Min(-jitter, maxJitterCorrectionPerSample);
+                            }
+                        }
+                    }
+                }
+            });
+        }
+
+        /// <summary>
+        /// Reads the current ADC result value on the selected channel
+        /// </summary>
+        /// <returns>24-bit raw value of the last ADC result (+ status byte if enabled)</returns>
+        public uint? ReadADCValue()
+        {
+            uint raw = GetRegisterValue(Register.Data);
+
+            // update the status register cache if we have it here
+            if (AppendStatusRegisterToData)
+            {
+                _registerCache[(byte)Register.Status] = (byte)(raw & 0xFF);
+                raw = (raw & 0xFFFFFF00) >> 8;
+            }
+
+            // check if we have an error
+            if (HasErrors)
+            {
+                return null;
+            }
+
+            // create the new AdcValue object and calculate the voltage
+            var adcValue = new AdcValue() { Raw = raw, Time = _stopWatch.ElapsedTicks, Channel = (byte)(GetRegisterBits(Register.Status, BitMask.StatusCHD)), Voltage = RawValueToVoltage(raw) };
+
+            // add it to the collection
+            // AdcValues.Enqueue(adcValue);
+            _adcValues.Add(adcValue);
+
+            // call the event handler
+            OnValueReceived?.Invoke(this, new AdcValueReceivedEventArgs(adcValue));
+
+            return raw;
+        }
+
+        /// <summary>
+        /// Reads a single value on the selected channel
```

Comment:
can you document how is this different from ReadADCValue

Reply:
Done